lib/models/messages/auxiliary/realtimecash.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:flutter/material.dart';
+import 'package:rtchat/models/messages/message.dart';
+
+class SimpleRealtimeCashDonationEventModel extends MessageModel {
+  final AssetImage image;
+  final String assetName;
+  final double value;
+  final String hash;
+
+  const SimpleRealtimeCashDonationEventModel(
+      {required this.assetName,
+      required this.image,
+      required this.value,
+      required this.hash,
+      required String messageId,
+      required DateTime timestamp})
+      : super(messageId: messageId, timestamp: timestamp);
+
+  static SimpleRealtimeCashDonationEventModel fromDocumentData(
+      String messageId, Map<String, dynamic> data) {
+    return SimpleRealtimeCashDonationEventModel(
+        assetName: data['activity']['asset'] ?? "UNKNOWN",
```

Comment:
```suggestion
  final double value;
  final String hash;

  const SimpleRealtimeCashDonationEventModel(
      {required this.image,
      required this.value,
      required this.hash,
      required String messageId,
      required DateTime timestamp})
      : super(messageId: messageId, timestamp: timestamp);

  static SimpleRealtimeCashDonationEventModel fromDocumentData(
      String messageId, Map<String, dynamic> data) {
    return SimpleRealtimeCashDonationEventModel(
```

`assetName` can be removed since it isn't used anywhere I think?

Reply:
Ohh, I see you're using `assetName` to mean currency. I would rename it to `currency` but up to you.